### PR TITLE
Event refactor

### PR DESCRIPTION
**Pull request by swalker2m**
Collapses created and updated events into one.  Simplifies logic surrounding inserts.  This is a breaking change.

### CODE REVIEW

File: modules/core/src/main/scala/lucuma/odb/api/model/AsterismModel.scala [hunk 1]
```diff
@@ -207,26 +207,17 @@ object AsterismModel extends AsterismOptics {
...
-  }
-
-  final case class AsterismEditedEvent (
+  final case class AsterismEvent (
     id:       Long,
-    oldValue: AsterismModel,
-    newValue: AsterismModel
-  ) extends Event.Edited[AsterismModel]
+    editType: Event.EditType,
+    value:    AsterismModel,
```
Comment (cquiroz) [inline modules/core/src/main/scala/lucuma/odb/api/model/AsterismModel.scala:213]: trailing comma

File: modules/core/src/main/scala/lucuma/odb/api/repo/TopLevelRepo.scala [hunk 2]
```diff
@@ -12,6 +12,7 @@ import cats.data._
 import cats.effect.concurrent.Ref
 import cats.syntax.apply._
 import cats.syntax.applicative._
+import cats.syntax.either._
```
Comment (cquiroz) [inline modules/core/src/main/scala/lucuma/odb/api/repo/TopLevelRepo.scala:15]: Any reason not to just import `cats.implicits._`?

**Review by cquiroz — APPROVED** [review]
I can't really comment on the details but it looks good and we discussed on the chat about removing the `oldValue` field

File: modules/core/src/main/scala/lucuma/odb/api/repo/TopLevelRepo.scala (hunk 2 above)
Comment (swalker2m) [reply modules/core/src/main/scala/lucuma/odb/api/repo/TopLevelRepo.scala:15]: Not really.  I just had a vague sense that a la carte imports are preferred.  What is the consensus?

File: modules/core/src/main/scala/lucuma/odb/api/model/AsterismModel.scala (hunk 1 above)
Comment (swalker2m) [reply modules/core/src/main/scala/lucuma/odb/api/model/AsterismModel.scala]: Thanks. :+1:

**Pull request merged**